File: examples/test_js_project/calculator.js
/**
 * Класс калькулятора для выполнения математических операций
 */
class Calculator {
    /**
     * Сложение двух чисел
     * @param {number} a первое число
     * @param {number} b второе число
     * @returns {number} сумма чисел
     */
    add(a, b) {
        return a + b;
    }

    /**
     * Вычитание двух чисел
     * @param {number} a первое число
     * @param {number} b второе число
     * @returns {number} разность чисел
     */
    subtract(a, b) {
        return a - b;
    }

    /**
     * Умножение двух чисел
     * @param {number} a первое число
     * @param {number} b второе число
     * @returns {number} произведение чисел
     */
    multiply(a, b) {
        return a * b;
    }

    /**
     * Деление двух чисел
     * @param {number} a делимое
     * @param {number} b делитель
     * @returns {number} частное
     * @throws {Error} если делитель равен нулю
     */
    divide(a, b) {
        if (b === 0) {
            throw new Error('Деление на ноль невозможно');
        }
        return a / b;
    }
}

module.exports = Calculator;

File: examples/test_python_project/calculator.py
"""
Модуль калькулятора для выполнения математических операций
"""


class Calculator:
    """Класс калькулятора для демонстрации функциональности"""
    
    def add(self, a: float, b: float) -> float:
        """
        Сложение двух чисел
        
        Args:
            a: первое число
            b: второе число
            
        Returns:
            сумма чисел
        """
        return a + b
    
    def subtract(self, a: float, b: float) -> float:
        """
        Вычитание двух чисел
        
        Args:
            a: первое число
            b: второе число
            
        Returns:
            разность чисел
        """
        return a - b
    
    def multiply(self, a: float, b: float) -> float:
        """
        Умножение двух чисел
        
        Args:
            a: первое число
            b: второе число
            
        Returns:
            произведение чисел
        """
        return a * b
    
    def divide(self, a: float, b: float) -> float:
        """
        Деление двух чисел
        
        Args:
            a: делимое
            b: делитель
            
        Returns:
            частное
            
        Raises:
            ValueError: если делитель равен нулю
        """
        if b == 0:
            raise ValueError("Деление на ноль невозможно")
        return a / b